src/Findex.tsx

```diff
@@ -0,0 +1,251 @@
+      </Stack>
+      <Stack spacing={3}>
+        {/* CREATE FINDEX KEY */}
+        <HeadingWithCode heading="Generate Findex key" code="/src/actions/createFindexKey.ts" />
+        <Text>Findex uses a single symmetric 128 bit key to upsert and search. Encryption is performed using AES 128 GCM.</Text>
+        <CodeHigligter codeInput={code?.createFindexKey} />
+        <Button onClick={handleCreateFindexKey} width="100%">
+          Create Findex key
+        </Button>
+        {findexKey && (
+          <Center gap="2">
+            <CheckCircleIcon color="green.500" />
+            Findex key created
+          </Center>
+        )}
+      </Stack>
+      <Stack spacing={3}>
+        {/* CREATE FINDEX LABEL */}
+        <HeadingWithCode heading="Labeling: salting the encryption" code="/src/actions/defineLabel.ts" />
+        <Text>
+          When indexing, the encryption uses an arbitrarily chosen public label; this label may represent anything, such as a period, e.g.,
+          “Q1 2022”. It should be changed when the index is compacted or recreated. Changing it regularly significantly increases the
+          difficulty of performing statistical attacks.
+        </Text>
+        <CodeHigligter codeInput={code?.defineLabel} />
+        <Button onClick={handleDefineLabel} width="100%">
+          Define Label
+        </Button>
+        {label && (
+          <Center gap="2">
+            <CheckCircleIcon color="green.500" />
+            Label defined
+          </Center>
+        )}
+      </Stack>
+      <Stack spacing={3}>
+        {/* DEFINE CALLBACKS*/}
+        <HeadingWithCode heading="Define callbacks" code="/src/actions/defineCallbacks.ts" />
+        <Text>
+          The Findex library abstracts the calls to the tables hosting the indexes. The developer is expected to provide the database’s
+          backend, typically a fast key/value store, and implement the necessary code in the callbacks used by Findex. Findex uses two
+          tables: the Entry table and the Chain table. Both have two columns: uid and value.
+        </Text>
+        <Text>
+          {" "}
+          To keep it simple here, we can use default callbacks available in cloudproof_js to fetch and upstert elements in both tables - in
+          memory.
+        </Text>
+        <CodeHigligter codeInput={code?.defineCallbacks} />
+      </Stack>
+      <Stack spacing={3}>
+        {/* UPSERT DATA */}
+        <HeadingWithCode heading="Index database" code="/src/actions/upsertData.ts" />
+        <Text>
+          To perform insertions or updates (a.k.a upserts), supply an array of IndexedEntry. This structure maps an IndexedValue to a list
+          of Keywords.
+        </Text>
+        <Text>Its definition is :</Text>
+        <CodeHigligter codeInput={code?.upsertData} />
+        <Button onClick={handleUpsertData} width="100%" isDisabled={!findexKey || !label}>
+          Index database
+        </Button>
+      </Stack>
+      <Stack spacing={3}>
```

Comment:
Add a copy of the indexed table somewhere in introduction or "Index database" bloc?